arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1746,87 +1746,49 @@ void RestReplicationHandler::handleCommandRestoreView() {
   LOG_TOPIC(TRACE, Logger::REPLICATION) << "restoring view: "
     << nameSlice.copyString();
 
-  if (ServerState::instance()->isCoordinator()) {
-    try {
-      auto* ci = ClusterInfo::instance();
-      auto view = ci->getView(_vocbase.name(), nameSlice.toString());
-
-      if (view) {
-        if (overwrite) {
-          auto res = view->drop();
+  try {
+    CollectionNameResolver resolver(_vocbase);
+    auto view = resolver.getView(nameSlice.toString());
 
-          if (!res.ok()) {
-            generateError(res);
+    if (view) {
+      if (overwrite) {
+        auto res = view->drop();
 
-            return;
-          }
-
-          generateError(TRI_ERROR_ARANGO_DUPLICATE_NAME);
+        if (!res.ok()) {
+          generateError(res);
 
           return;
         }
-      }
 
```

Comment:
only generate the duplicate name error here if _overwrite_ is false

Reply:
good catch